packages/gateway/src/features/restLocation/locationHandler.ts

```diff
@@ -188,37 +190,80 @@ export async function fetchLocationHandler(
   return response
 }
 
-function batchLocationsHandler(locations: Location[]) {
-  const locationsMap = new Map(
-    locations.map((location) => [
-      location.statisticalID,
-      { ...location, uid: `urn:uuid:${uuid()}` as URNReference }
-    ])
-  )
-  const locationsBundle = {
-    resourceType: 'Bundle',
-    type: 'document',
-    entry: locations
-      .map((location) => ({
-        ...location,
-        // partOf is either Location/{statisticalID} of another location or 'Location/0'
-        partOf:
-          locationsMap.get(location.partOf.split('/')[1])?.uid ??
-          location.partOf
-      }))
-      .map(
-        (location): BundleEntry<FhirLocation> => ({
-          fullUrl: locationsMap.get(location.statisticalID)!.uid,
-          resource: {
-            ...composeFhirLocation(location),
-            ...(location.statistics && {
-              extension: generateStatisticalExtensions(location.statistics)
-            })
-          }
-        })
+function createChunks<T>(array: T[], limit: number): T[][] {
+  const result = []
+  for (let i = 0; i < array.length; i += limit) {
+    result.push(array.slice(i, i + limit))
+  }
+  return result
+}
+
+function createLocationSegments(locations: Location[]): Location[][] {
+  const segments = []
+  for (const jurisdictionType of Object.keys(JurisdictionType)) {
+    const jurisdictionLocations = locations.filter(
+      (loc) => loc.jurisdictionType === jurisdictionType
+    )
+    if (jurisdictionLocations.length) {
+      segments.push(...createChunks(jurisdictionLocations, LOCATION_CHUNK_SIZE))
+    }
+  }
+  const facilitiesOrOffices = locations.filter((loc) => !loc.jurisdictionType)
+  if (facilitiesOrOffices.length) {
+    segments.push(...createChunks(facilitiesOrOffices, LOCATION_CHUNK_SIZE))
+  }
+  return segments
+}
```

Comment:
This logic is probably not needed anymore now that you implemented the length based approach, right?

Reply:
agreed, just putting the admin locations first would be sufficient

Reply:
Sorry, @rikukissa  I think it should stay the same, as I am updating the `statisticalToFhirIDMap` after processing each segment. If we remove this jurisdiction-type filtering logic, there will be a chance of staying parent and child locations in the same segment. Then the `fhirID` of the parent location cannot be resolved as the map will not already be updated

Reply:
Is it possible to first, right in the beginning the whole process, create the full parent map from all locations? @tahmidrahman-dsi

Reply:
I cannot think of any way of doing it 🤔 ... because I cannot get the `fhirID`s before sending them to hearth

Reply:
Annoying! Yea, you are right. It's not as easy as I thought. In that case this looks good to me 👍